.buildkite/dagster-buildkite/dagster_buildkite/package_spec.py

```diff
@@ -276,34 +282,44 @@ def distribution(self):
         setup = Path(self.directory) / "setup.py"
         if setup.exists():
             return run_setup(Path(self.directory) / "setup.py")
-        else:
-            return Distribution()
 
     @property
     def requirements(self):
-        extras = [
-            requirement
-            for requirements in self.distribution.extras_require.values()
-            for requirement in requirements
-        ]
-        install = self.distribution.install_requires
-        return extras + install
+        # First try to infer requirements from the distribution
+        if self.distribution:
+            extras = [
+                requirement
+                for requirements in self.distribution.extras_require.values()
+                for requirement in requirements
+            ]
+            install = self.distribution.install_requires
+            return extras + install
+
+        # If we don't have a distribution (like many of our integration test suites)
```

Comment:
Another option is to turn these into actual distributions? I don't think that's the right thing to do though.

At some point, we might want to source all of our dependencies from the same file/type of file anyway and then have setup.py read from/generated from there. That's how several of the build tools I've been looking into work. But I think adding the lightweight requirements.txt for now is fine.